[게시판 회원 기능] 에단 미션 제출합니다

Laterality opened the pull request:
안녕하세요, 이번에 미션 리뷰를 받게 된 에단입니다.

로그인 상태 유지를 구현하는 데 세션을 사용하면서 테스트가 다소 복잡하게 작성된 것 같습니다. 아직 `WebTestClient` 사용에 익숙하지 않아서 그렇다는 생각도 드는데, 개선할 수 있는 부분이 있을지 살펴봐주시면 큰 도움이 될 것 같습니다!

감사합니다.

Laterality committed:
Move template files

- Added thymeleaf dependency.

Laterality committed:
Add README

Laterality committed:
Add example code

Laterality committed:
Implement navigation between index page and article edit page

Laterality committed:
Implement article publish request

Laterality committed:
Implement saving article

Laterality committed:
Implement article list in main page

Laterality committed:
Implement article view

Laterality committed:
Edit README

Laterality committed:
Implement update article

Laterality committed:
Implement delete article

Laterality committed:
Enable live reload resource

Laterality committed:
Refactor redirection

 - Changed Article field name

Laterality committed:
Segregate common resource tags

Laterality committed:
Correct typo

Laterality committed:
Correct typo

Laterality committed:
Edit web ui home icon's href location

Laterality committed:
Add test code follow redirection

Laterality committed:
Apply feedbacks

Laterality committed:
Add test for delete controller

Laterality committed:
Changed to use constructor injection

Laterality committed:
Update web ui

Laterality committed:
Apply feedbacks

- Added method for update article.
- Segregated request dto from domain one.

Laterality committed:
Add Dto for article request

Laterality committed:
Refactor `ArticleRequestDto`

- Changed not to use setter in static factory method.

Laterality committed:
Refactor `Article` class

- Removed setters.

Laterality committed:
Update build.gradle

Laterality committed:
Update to use JPA

- Refactored controller test not to use repository directly.

Laterality committed:
Update README with new requirement(2nd week - step 1)

Laterality committed:
Update README typo

Laterality committed:
Update README with missed requirements

Laterality committed:
Implement registering user

- Refactored article service to use `save` instead of `update`.

Laterality committed:
Implement form validation of web ui

Laterality committed:
Update web ui

- Applied model to user list template.
- Changed link.

Laterality committed:
Update properties to show sql

Laterality committed:
Add dependency for MySql JDBC

- Added properties for test environment db.

Laterality committed:
Changed duplicate case input

Laterality committed:
Update signup view

- Added missed attributes.

Laterality committed:
Update README with new requirements (step 2)

Laterality committed:
Implement user login

- Chnaged fragment file names.
- Applied fragments to other templates.
- Segregated web ui header to fragment.

Laterality committed:
Implement redirection when user signed in try to move login/signup page

Laterality committed:
Implement check whether user is signed in when move to mypage

Laterality committed:
Changed README

Laterality committed:
Update README with new requirements(step 3)

Laterality committed:
Implement navigate to mypage edit view

- Added button for user withrawal.

Laterality committed:
Implement update user with web view

- Passed hand-test but not `UserControllerTest#user_edit`.

Laterality committed:
Add interceptor for logging

Laterality committed:
Update README

- Removed tests in user controller test related to login.

Laterality committed:
Implement user withdrawal

Laterality committed:
Update README

Laterality committed:
Refactor: segregate dto to extra package

Laterality committed:
Refactor: segregate commonly used methods and constant

Laterality committed:
Merge branch 'feat/interceptor' into Laterality

Laterality committed:
Merge branch 'Laterality' of https://github.com/woowacourse/jwp-blog into Laterality

Conflicts:
	README.md
	build.gradle
	src/main/java/techcourse/myblog/domain/Article.java
	src/main/java/techcourse/myblog/domain/ArticleRepository.java
	src/main/java/techcourse/myblog/service/ArticleService.java
	src/main/java/techcourse/myblog/web/ArticleController.java
	src/main/resources/application.properties
	src/main/resources/templates/article-edit.html
	src/main/resources/templates/article.html
	src/main/resources/templates/index.html
	src/main/resources/templates/mypage-edit.html
	src/main/resources/templates/mypage.html
	src/test/java/techcourse/myblog/domain/ArticleRepositoryTest.java
	src/test/java/techcourse/myblog/web/ArticleControllerTests.java

Changes to be committed:
	modified:   src/main/java/techcourse/myblog/domain/Article.java
	modified:   src/main/java/techcourse/myblog/domain/User.java
	modified:   src/main/java/techcourse/myblog/web/ArticleController.java
	modified:   src/main/java/techcourse/myblog/web/UserController.java
	modified:   src/main/resources/templates/article.html
	modified:   src/main/resources/templates/index.html
	modified:   src/main/resources/templates/mypage-edit.html
	modified:   src/main/resources/templates/mypage.html
	modified:   src/test/java/techcourse/myblog/domain/ArticleRepositoryTest.java
	modified:   src/test/java/techcourse/myblog/web/ArticleControllerTests.java

Laterality committed:
Resolve failed tests

Laterality committed:
Refactor article entity not to use static variable to generate id

Laterality committed:
Implement request header-based authorization

Laterality committed:
Remove `ArticleService`

- Formatted code.

Laterality committed:
Update view template: missed during merging

Laterality committed:
Update view: missed during merging

Laterality committed:
Update user controller tests not to use authorization header

- Removed `AuthInterceptor` not more needed.
- Formatted code.

Laterality committed:
Update user create/update: missed validation

Laterality committed:
Update logging

Laterality committed:
Refactor: remove no-needed method

Laterality committed:
Refactor controllers: Use `@SesstionAttribute`

Laterality committed:
Refactor

- Changed user's field name: `userName` → `username`.
- Segregated from `UserController` to `UserService`.

Laterality committed:
Refactor: organize view markup

Laterality committed:
Refactor: move repository classes into extra package

kang-hyungu reviewed (changes requested):
안녕하세요 에단님 리뷰어 강현구입니다.
전체적으로 코드 깔끔하게 잘 작성하셨네요 :+1: 
회원 기능 개발하신 코드 확인하고 피드백 남겼어요
확인해보시고 반영해보아요!

Laterality committed:
Apply validation annotation

Laterality committed:
Apply `@Transactional`

- Added alert to mypage view.
- Chnaged user update and authenticate logic.
- Specified id generation strategy to `Article` entity.

Laterality committed:
Remove unnecessary code

Laterality committed:
Rename `ControllerUtil` to `LoginUtil`

Laterality committed:
Change as of renaming `LoginUtil`

Laterality committed:
Update `UserController` to use `@Valid`

Laterality committed:
Update `User` not to depend on repository

Laterality committed:
Refactor exception throwing

Laterality committed:
Refactor controllers not to use `Optional<User>` as a parameter of cotroller method

Laterality committed:
Refactor: Segregate putting user object into model from session

Laterality committed:
Remove unnecessary comment

Laterality committed:
Refactor to use `@Valid` instead of try-catch

Laterality committed:
Refactor email duplication check to use `@Valid` annotation

Laterality committed:
Refactor: remove user registration code no more needed

Laterality committed:
Refactor `UserRepositoryTest`

Laterality committed:
Refactor

- Removed unnecessary test.
- Extracted validation message.

Laterality commented:
리뷰 감사합니다!

어노테이션 기반의 validation으로 변경하면서 코드가 많이 개선된 것이 느껴졌습니다!

~~이 과정에서 궁금한 점이 하나 생겼는데요. 외부의 Service나 Repository에 의존하는 커스텀 validation을 추가해봤는데, 프로덕션 코드에서는 잘 동작하는데 테스트 환경에서는 `@Autowired` 객체가 주입되지 않아섯 `UserRequestDto`의 테스트가 실패하고 있습니다. 혹시 이런 형태의 validation 어노테이션은 다른 방식으로 테스트해야 하나요?~~
해결했습니다. `Validator`를 `@Autowired`로 주입받아야 하는군요!

Laterality committed:
Resolve custom validation test issue

Laterality committed:
Refactor `UniqueEmailValidator` to use constructor injection

Laterality committed:
Refactor to use constructor injection

Laterality committed:
Refactor user controller test

Laterality committed:
Merge branch 'Laterality' of https://github.com/Laterality/jwp-blog into Laterality

kang-hyungu reviewed (approved):
구현하신 내용 확인했어요!
코드 개선도 많이 됐고 여러 가지로 고민하시고 반영하신 모습이 좋네요 :clap: 
pr건은 머지 처리할게요! 구현하느라 수고하셨어요
